File: src/Application Specific/Classes/Entity Collections/Jobs/JobItem/JobItem.cs
﻿using System.Collections;

namespace FSM.Entity
{
    namespace JobItems
    {
        public class JobItem
        {
            private DataTypeValidator _dataTypeValidator;

            public JobItem()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            private int _JobItemID = 0;

            public int JobItemID
            {
                get
                {
                    return _JobItemID;
                }
            }

            public object SetJobItemID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobItemID", value);
                }
            }

            private int _JobOfWorkID = 0;

            public int JobOfWorkID
            {
                get
                {
                    return _JobOfWorkID;
                }
            }

            public object SetJobOfWorkID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_JobOfWorkID", value);
                }
            }

            private string _Summary = string.Empty;

            public string Summary
            {
                get
                {
                    return _Summary;
                }
            }

            public object SetSummary
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Summary", value);
                }
            }

            private int _RateID = 0;

            public int RateID
            {
                get
                {
                    return _RateID;
                }
            }

            public object SetRateID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_RateID", value);
                }
            }

            private decimal _Qty;

            public decimal Qty
            {
                get
                {
                    return _Qty;
                }
            }

            public object SetQty
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Qty", value);
                }
            }

            private int _systemLinkId = 0;

            public int SystemLinkID
            {
                get
                {
                    return _systemLinkId;
                }
            }

            public object SetSystemLinkID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_systemLinkId", value);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/JobItem/JobItem.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace JobItems

        Public Class JobItem

            Private _dataTypeValidator As DataTypeValidator

            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

#End Region

#Region "JobItem Properties"

            Private _JobItemID As Integer = 0

            Public ReadOnly Property JobItemID() As Integer
                Get
                    Return _JobItemID
                End Get
            End Property

            Public WriteOnly Property SetJobItemID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobItemID", Value)
                End Set
            End Property

            Private _JobOfWorkID As Integer = 0

            Public ReadOnly Property JobOfWorkID() As Integer
                Get
                    Return _JobOfWorkID
                End Get
            End Property

            Public WriteOnly Property SetJobOfWorkID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_JobOfWorkID", Value)
                End Set
            End Property

            Private _Summary As String = String.Empty

            Public ReadOnly Property Summary() As String
                Get
                    Return _Summary
                End Get
            End Property

            Public WriteOnly Property SetSummary() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Summary", Value)
                End Set
            End Property

            Private _RateID As Integer = 0

            Public ReadOnly Property RateID() As Integer
                Get
                    Return _RateID
                End Get
            End Property

            Public WriteOnly Property SetRateID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_RateID", Value)
                End Set
            End Property

            Private _Qty As Decimal

            Public ReadOnly Property Qty() As Decimal
                Get
                    Return _Qty
                End Get
            End Property

            Public WriteOnly Property SetQty() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Qty", Value)
                End Set
            End Property

            Private _systemLinkId As Integer = 0

            Public ReadOnly Property SystemLinkID() As Integer
                Get
                    Return _systemLinkId
                End Get
            End Property

            Public WriteOnly Property SetSystemLinkID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_systemLinkId", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace